be/src/storage/sstable/table_builder.cpp

```diff
@@ -142,6 +143,43 @@ static bool Snappy_Compress(const char* input, size_t length, ::std::string* out
     return true;
 }
 
+/**
+ * Compresses the input data using the specified compression type.
+ *
+ * @param compression_type The type of compression to use.
+ * @param input The input data to compress.
+ * @param length The length of the input data.
+ * @param output The output string to store the compressed data.
+ * @return Status::OK() if the data was compressed successfully, otherwise an error status.
+ */
+static Status CompressBlock(CompressionTypePB compression_type, const char* input, uint64_t length,
+                            std::string* output) {
+    // Get the appropriate compression codec for the specified compression type.
+    const BlockCompressionCodec* codec = nullptr;
+    RETURN_IF_ERROR(get_block_compression_codec(compression_type, &codec));
+
+    // Calculate the maximum length of the compressed data.
+    size_t output_length = codec->max_compressed_len(length);
+
+    // Resize the output string to accommodate the uncompressed length and the maximum compressed data length.
+    output->resize(sizeof(uint64_t) + output_length);
```

Comment:
should use `reserve` instead of `resize`?

Reply:
After test, we can't use reserve instead of resize here, because if we do that, the second resize will refill this output with invalid char, and decompress will fail